src/span-skip-list.coffee

```diff
@@ -11,11 +9,13 @@ class SpanSkipList
   #   A list of strings naming the dimensions to be indexed. Elements should
   #   have numeric-valued properties matching each of the indexed dimensions.
   constructor: (@dimensions...) ->
-    @head = new Node(@maxHeight, @buildZeroDistance())
-    @tail = new Node(@maxHeight, @buildZeroDistance())
-    for i in [0...@maxHeight]
-      @head.next[i] = @tail
-      @head.distance[i] = @buildZeroDistance()
+    @head = @createNode(@maxHeight, @buildZeroDistance())
+    @tail = @createNode(@maxHeight, @buildZeroDistance())
+    index = 0
+    while index < @maxHeight
```

Comment:
What does this return at the end of a method? It doesnt make a new array, right?